smart_contracts/bond_maturation/src/lib.rs

```diff
@@ -76,16 +78,46 @@ fn main(id: TriggerId, issuer: AccountId, event: Event) {
                 issued_bond.id().account_id()
             ));
         } else {
+            if buyer == issuer {
+                info!(&format!("{bond_id}: Buyer is the issuer, skipping maturity payment"));
+
+                continue;
+            }
+
             let bond_issuer_money = AssetId::new(bond_currency.clone(), issuer.clone());
 
-            TransferExpr::new(bond_issuer_money, amount, buyer)
+            info!(&format!(
+                "{bond_id}: Transferring {amount} {bond_currency} from {issuer} to {buyer}"
+            ));
+
+            TransferExpr::new(bond_issuer_money.clone(), amount.clone(), buyer.clone())
                 .execute()
-                .dbg_expect("Sending money failed. Country might have went bankrupt");
-        }
+                .dbg_expect("Sending money failed. Country might have gone bankrupt");
 
-        info!(&format!("{bond_id}: Bond matured"));
+            let transfer_metadata_id: Name = format!("maturity_payment_{}", bond_id.name().to_string())
```

Comment:
```suggestion
            let transfer_metadata_id: Name = format!("maturity_payment_{}", bond_id.name())
```

Reply:
```suggestion
            let transfer_metadata_id: Name = format!("maturity_payment%%{}%%{}", bond_id.name(), bond_id.domain_id())
```

I would recommend you use both `name` and `domain_id`. There could be another asset of the same name in another domain

Reply:
Impossible to have it in Palau project since there is only one domain, and each bond issue is validated before submitting to Iroha2. But yes, will add a domain just in case.